REPL.cpp

```diff
@@ -31,18 +30,30 @@ swift::FuncDecl *ConfigureFunctionLinkage(swift::SourceFile &src_file, std::uniq
 
             sil_module->lookUpFunction(sil_decl)->setLinkage(swift::SILLinkage::Public);
             Log(std::string("Set function ") + name_original + " (" + name_mangled + ") to public");
-
-            if(name_original == res_fn_name)
-                res_fn = fn;
         }
     }
     sil_module->lookUpFunction("main")->setLinkage(swift::SILLinkage::Private);
-    return res_fn;
 }
 
-llvm::Expected<std::unique_ptr<REPL>> REPL::Create()
+void REPL::RemoveRedeclarationsFromJIT(std::unique_ptr<llvm::Module> &llvm_module)
+{
+    SetCurrentLoggingArea(LoggingArea::SIL);
+
+    for(llvm::Function &fn : llvm_module->functions())
+    {
+        if(fn.isDeclaration() || !fn.hasExternalLinkage())
+            continue;
+        std::string name = fn.getName().str();
+        llvm::consumeError(m_jit->RemoveSymbol(name));
+
+        if(m_fn_ptr_map.find(name) == m_fn_ptr_map.end())
+            m_fn_ptr_map[name] = "";
+    }
+}
+
+llvm::Expected<std::unique_ptr<REPL>> REPL::Create(bool is_playground)
 {
-    std::unique_ptr<REPL> result(new REPL);
+    std::unique_ptr<REPL> result(new REPL(is_playground));
```

Comment:
Use `llvm::make_unqiue` instead?

Reply:
Constructor is protected, so doesn't work.